### ISSUE TITLE
New Game/Generator Request - Zelda: Tears of the Kingdom

### ISSUE THREAD

**Issue by McBraggins** [kbuzsaki/bingosync#298]
A list of possible goals for a TOTK bingo board. I know it's a lot, but you have to have some variety

Obtain Master Sword
Obtain the Hylian Shield
Obtain the Champion’s Tunic
Obtain the Well-Worn Hairband
‘Tulin of Rito Village’ Main Quest
‘Sidon of the Zora’ Main Quest
‘Yunobo of Goron City’ Main Quest
‘Riju of Gerudo Town’ Main Quest
Activate 15 Lightroots
Activate 14 Lightroots
Activate 13 Lightroots
Activate 12 Lightroots
Activate 11 Lightroots
Activate 10 Lightroots
Activate 9 Lightroots
Activate 8 Lightroots
Activate 7 Lightroots
Activate 6 Lightroots
Activate 5 Lightroots
Activate 30 Shrines
Activate 25 Shrines
Activate 20 Shrines
Activate 19 Shrines
Activate 18 Shrines
Activate 17 Shrines
Activate 16 Shrines
Activate 15 Shrines
Activate 14 Shrines
Activate 13 Shrines
Activate 12 Shrines
Activate 11 Shrines
Activate 10 Shrines
Activate 9 Shrines
Activate 8 Shrines
Activate 7 Shrines
Activate 6 Shrines
Activate 5 Shrines
Complete 20 Shrines
Complete 19 Shrines
Complete 18 Shrines
Complete 17 Shrines
Complete 16 Shrines
Complete 15 Shrines
Complete 14 Shrines
Complete 13 Shrines
Complete 12 Shrines
Complete 11 Shrines
Complete 10 Shrines
Complete 9 Shrines
Complete 8 Shrines
Complete 7 Shrines
Compete 6 Shrines
Complete 5 Shrines
Activate 15 Skyview Towers
Activate 14 Skyview Towers
Activate 13 Skyview Towers
Activate 12 Skyview Towers
Activate 11 Skyview Towers
Activate 10 Skyview Towers
Activate 9 Skyview Towers
Activate 8 Skyview Towers
Activate 7 Skyview Towers
Activate 6 Skyview Towers
Activate 5 Skyview Towers
Activate 4 Skyview Towers
Take a Selfie with a Lynel
Take a Selfie with Hestu
Take a Selfie with Beedle
Take a Selfie with Tulin
Take a Selfie with Sidon
Take a Selfie with Yunobo
Take a Selfie with Riju
Give a Beetle to Beedle
Complete 1 Temple Main Quest
Complete 2 Temple Main Quests
Complete 3 Temple Main Quests
Complete 4 Temple Main Quests
3 Shrines from -Reigon-
45 Koroks
40 Koroks
35 Koroks
30 Koroks
25 Koroks
20 Koroks
15 Koroks
10 Koroks
Kill 5 Like Likes
Kill 4 Like Likes
Kill 3 Like Likes
Kill an Electric Like Like
Kill a Fire Like Like
Kill an Ice Like Like
Kill a Lynel
Kill a Gloom Lynel
Kill a Obsidian Frox
Kill 5 Boss Bokoblins
Kill a Gleeok
Discover 6 Stables
‘Impa and the Geoglyphs’ Main Quest
‘Camera in the Depths’ Main Quest
Beat Ganon
Upgrade the Energy Cell Twice
‘Master Kohga of the Yiga Clan’ Side Adventures
Collect 5 Oil Jars
Attach a Korok to a Rocket and activate it
Complete 10 Buddy Koroks
Complete 9 Buddy Koroks
Complete 8 Buddy Koroks
Complete 7 Buddy Koroks
Complete 6 Buddy Koroks
Complete 5 Buddy Koroks
Change your Paraglider Design
Get Hit by a Falling Sky Island Rock
Catch a Star Fragment in the Sky
Obtain the Climber’s Set
Obtain Autobuild
‘Master Kohga of the Yiga Clan’ Side Adventure

**Comment by Alive527** [comment]
i know very little about LOZ, especially TOTK, but it looks like this board would be alot better in slrv5 so you dont get an overwhelming amount of synchronized goals, like activating or completing shrines, thought id let you know just incase

**Comment by BrandonEcholsMusic** [comment]
@McBraggins I have actually been working on a list of TotK Goals to request a Bingosync generator, and was wondering if my goals could be used instead. I feel like there needs to be multiple difficulties like the BotW one and even more variety. Please let me know if you're fine with this.

Thanks, 
      Brandon Echols

**Comment by BrandonEcholsMusic** [comment]
> > > i know very little about LOZ, especially TOTK, but it looks like this board would be alot better in slrv5 so you dont get an overwhelming amount of synchronized goals, like activating or completing shrines, thought id let you know just incase
> > 
> > 
> > What do you mean by slrv5? I know that the BOTW board just goals like "6 Shrines" and "10 Shrines" But I didn't know if it meant complete them or register them as travel points. I can get rid of the "Discover Shrines" because I feel like that makes more sense. But I still don't know what slrv5 is.
> 
> srlv5 sorts out all goals into 25 "sections" to fill out all 25 tiles. This means each tile can be one of however many goals you have. So one section of the srlv5 list could be all of the shrine goals i mentioned so that there would not be multiple of them. You also assign goal types to each goal in srlv5, so if you listed the shrine goal type as "shrine", then there would not ever be two goals with the "shrine" type in the same lines.
> 
> What you listed was a basic random list, where there is no sorting method to the goals, and bingosync just picks 25 random goals and randomly assorts them.

The BotW bingosync wasn't slrv5, and you would _rarely_ get duplicates of the same thing. With the list that I made, there's plenty of variation, so it would work fine with just a basic list.

**Comment by Alive527** [comment]
> > > > i know very little about LOZ, especially TOTK, but it looks like this board would be alot better in slrv5 so you dont get an overwhelming amount of synchronized goals, like activating or completing shrines, thought id let you know just incase
> > > 
> > > 
> > > What do you mean by slrv5? I know that the BOTW board just goals like "6 Shrines" and "10 Shrines" But I didn't know if it meant complete them or register them as travel points. I can get rid of the "Discover Shrines" because I feel like that makes more sense. But I still don't know what slrv5 is.
> > 
> > 
> > srlv5 sorts out all goals into 25 "sections" to fill out all 25 tiles. This means each tile can be one of however many goals you have. So one section of the srlv5 list could be all of the shrine goals i mentioned so that there would not be multiple of them. You also assign goal types to each goal in srlv5, so if you listed the shrine goal type as "shrine", then there would not ever be two goals with the "shrine" type in the same lines.
> > What you listed was a basic random list, where there is no sorting method to the goals, and bingosync just picks 25 random goals and randomly assorts them.
> 
> The BotW bingosync wasn't slrv5, and you would _rarely_ get duplicates of the same thing. With the list that I made, there's plenty of variation, so it would work fine with just a basic list.

botw list was srlv5,  bingosync-app/generators/goal_lists/zelda_botw_english.js
notice the "bingolist[1]" part, those are the sections. and every goal has a name and type, thats srlv5

**Comment by BrandonEcholsMusic** [comment]
@Alive527 Ooooooh, ok. That makes a lot of sense now. So I would just have to have 25 lists, and then it would choose one goal from each list, and then put them in random spaces on the board?

**Comment by Alive527** [comment]
> @Alive527 Ooooooh, ok. That makes a lot of sense now. So I would just have to have 25 lists, and then it would choose one goal from each list, and then put them in random spaces on the board?

correct, and this would help organize the board to jot have overlapping goals, and make everything more balanced. it is harder to make then just a list of goals, but does make the board more interesting

**Comment by BrandonEcholsMusic** [comment]
@Alive527 Ok... So I would just have to provide 25 lists?

**Comment by Alive527** [comment]
yes, i wouldn't call them their own list exactly, they dont have to be nearly as big as botw. I made a siege and undertale list in srlv5 and each sub group had like 4-5 goals, so id say just tru to organize your current goals well enough

**Comment by BrandonEcholsMusic** [comment]
@Alive527 But, if I wanted to, I could make each list have approx. 20 goals for _true_ randomization, right? Or would that be too hard to implement...

**Comment by BrandonEcholsMusic** [comment]
@Alive527 Because, if it _is_ really complicated, I don't mind learning the coding myself... (It can't be THAT hard, right?)

**Comment by Alive527** [comment]
if you had enough goals, your more than welcome to, the more the better, and even better with proper types to sort them out better

**Comment by BrandonEcholsMusic** [comment]
@Alive527 Ok. Thanks! So, do I need permission from @McBraggins for my lists to be implemented, or is this thread for someone (i.e. _me_) to come up with the lists.

**Comment by Alive527** [comment]
ima be honest i thought you were them for a sec, this thread is for their list to be implemented, but you could talk them into doing the srlv5 version for kbuzsaki to upload, or maybe ask kbuz to add both and label them normal and srlv5 if there is disagreement, thats up to Mcbraggins tho

**Comment by BrandonEcholsMusic** [comment]
@Alive527 Ok.

**Comment by BrandonEcholsMusic** [comment]
@McBraggins Hi! As you can see in the previous conversation, I've been speaking with **Alive527** about having an slrv5 version of this generator. I've been working pretty hard on a list with lots of variation. I've attached a link to a spreadsheet if you want to take a look. I feel like the variation in this will be very entertaining to play on the board. Please give me your opinions and let me know if you think my list should be implemented. **Alive527** also said that both of ours could be implemented, if we cannot agree. What do you think?

Brandon Echols

**Comment by akumar282** [comment]
Hey @BrandonEcholsMusic this list looks great. Do you mind if I get started on adding it?

**Comment by Alive527** [comment]
> Hey @BrandonEcholsMusic this list looks great. Do you mind if I get started on adding it?

if you do and know about the game, its sorted into 25 lists but it probably still needs types for srlv5 to properly sort them before being implemented onto the website

**Comment by akumar282** [comment]
> if you do and know about the game, its sorted into 25 lists but it probably still needs types for srlv5 to properly sort them before being implemented onto the website

Yep! I took a look at the documentation, as well as the botw implementation just to get an idea of how it was done there. I'll sort them and get started.

**Comment by haylau** [comment]
> @McBraggins Please give me your opinions and let me know if you think my list should be implemented. **Alive527** also said that both of ours could be implemented, if we cannot agree. What do you think?

The lists could always be merged then implemented in srlv5 I think I see a few that aren't on @McBraggins's list.

**Comment by haylau** [comment]
Some things that aren't in the current srlv5 list, lemme know what y'all think? I kinda just kept thinking of things..

‘Tulin of Rito Village’ Main Quest
‘Sidon of the Zora’ Main Quest
‘Yunobo of Goron City’ Main Quest
‘Riju of Gerudo Town’ Main Quest
each specific tower? 
‘Impa and the Geoglyphs’ Main Quest
demon king's army
demon king ganondorf
demon dragon
sleep in x beds 
x dog treasure
Upgrade energy cell once/twice -> obtain x energy cells (prefer the language/could increase the number)
obtain x poe
obtain a sleepover ticket
obtain earthwake manual
obtain a specific capsule?
cure great deku tree
obtain x old map
find x bargainer statues
clear x colloseums
majora's mask
complete x side quests?
complete x side adventures?

also could add but might be a little overkill
each specific equipment (some full sets are on here already)
upgrade specific equipment 
each specific shrine? (already has some shrine quests, might be too much idk)
each specific side quest? (already has a good set of them tho)
region koroks (never liked this one tho lmao)

**Comment by BrandonEcholsMusic** [comment]
> Some things that aren't in the current srlv5 list, lemme know what y'all think? I kinda just kept thinking of things..
> 
> ‘Tulin of Rito Village’ Main Quest ‘Sidon of the Zora’ Main Quest ‘Yunobo of Goron City’ Main Quest ‘Riju of Gerudo Town’ Main Quest each specific tower? ‘Impa and the Geoglyphs’ Main Quest demon king's army demon king ganondorf demon dragon sleep in x beds x dog treasure Upgrade energy cell once/twice -> obtain x energy cells (prefer the language/could increase the number) obtain x poe obtain a sleepover ticket obtain earthwake manual obtain a specific capsule? cure great deku tree obtain x old map find x bargainer statues clear x colloseums majora's mask complete x side quests? complete x side adventures?
> 
> also could add but might be a little overkill each specific equipment (some full sets are on here already) upgrade specific equipment each specific shrine? (already has some shrine quests, might be too much idk) each specific side quest? (already has a good set of them tho) region koroks (never liked this one tho lmao)

Just like BotW, I think there should be normal, short, and long varients. The spreadsheet currently just has the normal varient. By the end of tomorrow (maybe even earlier in the day) I should have all the varients done. Wait until then to implement them, as I might change some of the current things to fit.

**Comment by BrandonEcholsMusic** [comment]
> > @McBraggins Please give me your opinions and let me know if you think my list should be implemented. **Alive527** also said that both of ours could be implemented, if we cannot agree. What do you think?
> 
> The lists could always be merged then implemented in srlv5, I think I see a few that aren't on @McBraggins's list.

Just as I explained in my previous comment, all of the goals that @McBraggins proposed will be implemented in the other varients.

**Comment by BrandonEcholsMusic** [comment]
> Some things that aren't in the current srlv5 list, lemme know what y'all think? I kinda just kept thinking of things..
> 
> ‘Tulin of Rito Village’ Main Quest ‘Sidon of the Zora’ Main Quest ‘Yunobo of Goron City’ Main Quest ‘Riju of Gerudo Town’ Main Quest each specific tower? ‘Impa and the Geoglyphs’ Main Quest demon king's army demon king ganondorf demon dragon sleep in x beds x dog treasure Upgrade energy cell once/twice -> obtain x energy cells (prefer the language/could increase the number) obtain x poe obtain a sleepover ticket obtain earthwake manual obtain a specific capsule? cure great deku tree obtain x old map find x bargainer statues clear x colloseums majora's mask complete x side quests? complete x side adventures?
> 
> also could add but might be a little overkill each specific equipment (some full sets are on here already) upgrade specific equipment each specific shrine? (already has some shrine quests, might be too much idk) each specific side quest? (already has a good set of them tho) region koroks (never liked this one tho lmao)

@haylau  I will have most of these goals in the spreadsheet my tomorrow.

@Alive527 @akumar282 @haylau @McBraggins Before continuing with anything, wait until I confirm (in this github issue) that the lists are complete. Thank you for your understanding.

**Comment by haylau** [comment]
Sounds good. Let me know if you need any help!

 Looked at the srlv5 for botw and I don't think that long and short modes used separate lists, as I can't find them. It looks like the only difference between them is a mode setting for the srlv5 generator but I can't tell if each 'bucket' is supposed to have some inherent difficult value or something along those lines? Not really sure how it works under the hood as there's a lot of undocumented code.

**Comment by BrandonEcholsMusic** [comment]
@haylau If you go onto the Bingosync website, and choose BotW, you'll see the different variants. I'm not exactly sure how this is done, but I'm making separate lists anyway. This is able to be implemented, right?

**Comment by haylau** [comment]
Yeah, I've used the short/normal variants before, but it doesn't appear as they use separate lists, rather the generator works differently:

https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/zelda_botw_long_generator.js#L1-L7
https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/generator_bases/srl_generator_v5.js#L162-L166

**Comment by haylau** [comment]
We'd also need to add labels to goals, as the generator avoids placing similar ones together:

https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/goal_lists/zelda_botw_english.js#L2-L13

@akumar282 and I briefly started doing this but we'll wait for the new list

**Comment by haylau** [comment]
a newline might work too:

![image](https://github.com/kbuzsaki/bingosync/assets/13607064/3ade9c59-e805-4ed4-a560-6e0488c39b16)

**Comment by BrandonEcholsMusic** [comment]
@haylau So, can you please explain in detail what you need the spreadsheet to look like? I now understand that there should be a type column, right?

**Comment by haylau** [comment]
> @haylau So, can you please explain in detail what you need the spreadsheet to look like? I now understand that there should be a type column, right?

We'll only need one list containing all of the goals, separated into 25 'buckets' in order of "goal difficulty". Types for each goal should be on the following columns. Something like this:
![image](https://github.com/kbuzsaki/bingosync/assets/13607064/a339d7f8-72bf-44b1-bc70-8b6c28d6fba5)

**Comment by BrandonEcholsMusic** [comment]
@haylau So, what you're saying is that there just needs to be one long list split into 25 sections; the first section will contain goals of the easiest, while the 25th section will contain the hardest goals? Please explain more of what you need to the types.

**Comment by haylau** [comment]
Correct. For example, the first two from BoTW's board would look like
https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/goal_lists/zelda_botw_english.js#L2-L18
![image](https://github.com/kbuzsaki/bingosync/assets/13607064/cb4b252e-ac88-4a1a-9f46-68c8803cec44)
etc.

**Comment by BrandonEcholsMusic** [comment]
@haylau If you look on the "NEW" sheet on the spreadsheet, this is what you're talking about right?

**Comment by haylau** [comment]
yep!

**Comment by BrandonEcholsMusic** [comment]
Perfect! I'll get working on it right now. Thanks a lot!

**Comment by BrandonEcholsMusic** [comment]
I'll let you know when I'm done.

**Comment by haylau** [comment]
Want any help reorganizing? @akumar282 is bored. haha

**Comment by BrandonEcholsMusic** [comment]
Haha. If anyone wants to help, start putting numbers between 1 and 25 (based on the goal's difficulty) in column b. Then, I can sort by that column and we can copy and paste into the "NEW" sheet. I'll give editing access to anyone with the link. :)

**Comment by BrandonEcholsMusic** [comment]
To be clear, this is column B on the "Unused" sheet.

**Comment by BrandonEcholsMusic** [comment]
Also, feel free to add more goals on the bottom. Honestly, with this new format, I think we could have literally every quest, etc.

**Comment by haylau** [comment]
Yeah. Just a matter of actually adding them in considering how much content TotK has...

**Comment by BrandonEcholsMusic** [comment]
Yea. ACTUALLY, if we are to do something _like that,_ we should probably just start over on the "NEW" sheet so we can add more of the "x old maps" (for example) goals now that it's not sorted into short, normal, and long.

**Comment by haylau** [comment]
I also can't completely tell how the generator puts stuff together but BotW's buckets put a lot of similar goals in the same bucket but I also don't really know if that's optimal since it avoids similar typed goals in the same row/col/diag, just something to keep in mind? Since having a board having 4, 5, 6, 7, and 8 koroks as goals isn't ideal

**Comment by BrandonEcholsMusic** [comment]
Yea. So pretty much what we're doing here is the "EXAMPLE" sheet it just of former goals. We now need to put all the new goals (based off of those) in the "EDIT THIS SHEET" sheet. Once we all agree that we're done, I can sort it and we can put it in the NEW sheet. From there, we just add labels.

**Comment by BrandonEcholsMusic** [comment]
Just pay attention to column B on the EXAMPLE sheet. I'm pretty much writing down all the formats. Just start putting LOTS of these goals in the EDIT THIS SHEET sheet.

**Comment by BrandonEcholsMusic** [comment]
Now, anyone can start putting goals of the specified formats (on the EXAMPLE sheet) into Column A of the EDIT THIS SHEET sheet. Don't forget the goal's difficulty in column B.

**Comment by Alive527** [comment]
> Yeah, I've used the short/normal variants before, but it doesn't appear as they use separate lists, rather the generator works differently:
> 
> https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/zelda_botw_long_generator.js#L1-L7
> 
> 
> https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/generator_bases/srl_generator_v5.js#L162-L166

i have just discovered, i am pretty sure tears of the kingdom is NOT in srlv5 but srlv8 which is WAY more complex if i am correct, and i dont know myself how it works yet, which is why it can just take the values short and long

**Comment by Alive527** [comment]
> Actually, I think the buckets are being used as a form of difficulty? not quite sure how the generator operates as there's a lot of magic numbers all over the place but the goals in bucket 25 definitely look like are intended to be harder than the goals in bucket 1.

also the 25 buckets dont classify difficulty exactly, i think its just common for people to put harder goals in the later buckets but isnt required. This might be to not have the possibility of many hard goals happening if they are spread out among the buckets

**Comment by haylau** [comment]
> > Yeah, I've used the short/normal variants before, but it doesn't appear as they use separate lists, rather the generator works differently:
> > https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/zelda_botw_long_generator.js#L1-L7
> > 
> > https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/generator_bases/srl_generator_v5.js#L162-L166
> 
> i have just discovered, i am pretty sure tears of the kingdom is NOT in srlv5 but srlv8 which is WAY more complex if i am correct, and i dont know myself how it works yet, which is why it can just take the values short and long

hmm I made the assumption it was using v5 since it was using the v5 generator.




> > Actually, I think the buckets are being used as a form of difficulty? not quite sure how the generator operates as there's a lot of magic numbers all over the place but the goals in bucket 25 definitely look like are intended to be harder than the goals in bucket 1.
> 
> also the 25 buckets dont classify difficulty exactly, i think its just common for people to put harder goals in the later buckets but isnt required. This might be to not have the possibility of many hard goals happening if they are spread out among the buckets

Noted. The plan currently is to sort by difficulty for that exact reason: to not have the possibility of many hard goals happening, but was under impression that that's how the short and long variants worked?

We're finalizing the list right now if you want to look at it. https://docs.google.com/spreadsheets/d/1kZFM9KOi9AFq0gtx8ulqifrKaPHCNqdjNh9le56qxis/edit?pli=1#gid=420452587

**Comment by haylau** [comment]
> i have just discovered, i am pretty sure tears of the kingdom is NOT in srlv5 but srlv8 which is WAY more complex if i am correct, and i dont know myself how it works yet, which is why it can just take the values short and long

It doesn't look like the BotW list uses the extra tags that v8 says it does:
https://github.com/kbuzsaki/bingosync/wiki/Generator-Bases-Documentation
https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/goal_lists/zelda_botw_english.js#L2-L18

**Comment by BrandonEcholsMusic** [comment]
@haylau I'm back! Is there a problem?

**Comment by BrandonEcholsMusic** [comment]
@haylau Please explain to me what's wrong.

**Comment by kbuzsaki** [comment]
SRL v8 is backwards compatible with goal lists that were made for v5, it just changes how the generator handles synergy overload and adds some extra logic that was used for OoT short bingo like child-only goals.

For bingosync goal lists it's not necessary to provide all three of short, normal, and long if you don't want them, it's fine to just provide one :)

I haven't been following along with this thread too closely, but I'm happy to put up whatever goal list y'all agree on, as long as it is in the correct json or javascript format. Please don't send me a spreadsheet, the generator code can't handle those :)

**Comment by BrandonEcholsMusic** [comment]
@kbuzsaki We're working on making a list in slrv5 format, with increasing difficulty throughout the 25 "buckets." We _want_ to have short, normal, and long varients of the game. I'm not exactly sure how this is going to be implemented (I'm _not_ a programming genius), but I want to make sure that having a list in slrv5 format will be able to be converted into short, normal, and long goal lists? Is this how it's going to work or do we need to format it differently, or with respective lists for short, normal, and long?

**Comment by haylau** [comment]
> For bingosync goal lists it's not necessary to provide all three of short, normal, and long if you don't want them, it's fine to just provide one :)

Are they intended to be separate lists? BotW offers short/normal/long boards but only uses one list? The code's a bit difficult to follow haha.

**Comment by BrandonEcholsMusic** [comment]
@haylau Yea, I gave it an attempt and it makes no sense... We _should_ be able to still get the short/normal/long from the one list right? (I hope we don't have to separate the list _again_...)

**Comment by haylau** [comment]
If it's necessary to generate three lists, since we'll have goals organized by difficulty, we can just map a subset of those goals for the short and long ones, but if it does require separate lists than I'm not quite sure how BotW has separate goal sets.

**Comment by BrandonEcholsMusic** [comment]
@haylau Before we continue on the spreadsheet, maybe we should figure out how/if we would be able to generate the short/normal/long varients from the one list... I'm sure @kbuzsaki would have a few ideas.

**Comment by BrandonEcholsMusic** [comment]
@haylau I've looked at all the code (at least I _think_), and the only thing that's added to `bingosync-app/generators/zelda_botw_short_generator.js` compared to the normal one is the following code:

`bingoGenerator = function(goals, opts) {
    opts.mode = "short";
    return _bingoGenerator(goals, opts);
};`

I'm not sure exactly what this means, but the long one just replaces the above "short" with "long." I'm not exactly a programming genius, but I have _no idea_ how it can call just the short goals with the above code. In the goal lists file for botw, there's nothing relating to difficulty or varients... Am I missing something here?

**Comment by haylau** [comment]
it seems that it passes the board card mode to the generator which is used for a difficulty function
https://github.com/kbuzsaki/bingosync/blob/48753c6ccabecb67a70a1289e185ffe33cd3a11f/bingosync-app/generators/generator_bases/srl_generator_v5.js#L151-L158, but i've no clue why it's used.

**Comment by BrandonEcholsMusic** [comment]
@haylau Ok... Hmmm. Can we pretty much just use the exact same code from the BotW board, but just replace the lists with our own lists? I feel like that wouldn't be too hard.

**Comment by haylau** [comment]
That was my plan 🤷. If we need to make separate short/long lists, it'll be much easier once we have the normal one finished.

**Comment by BrandonEcholsMusic** [comment]
@haylau Ok. We'll work on that tomorrow.

**Comment by haylau** [comment]
@kbuzsaki Will having too many goals be a problem?

**Comment by BrandonEcholsMusic** [comment]
@kbuzsaki @haylau We're planning on having ~3000 in the end.

**Comment by haylau** [comment]
My guess is 2300 lol

**Comment by BrandonEcholsMusic** [comment]
Yea, that's probably more realistic.

**Comment by haylau** [comment]
> My guess is 2300 :)

I was off by 26!

**Comment by BrandonEcholsMusic** [comment]
@haylau Nice job! That was fast. Is it actually implemented yet? I don't see it on the website yet... Does it take some time to update?

**Comment by McBraggins** [comment]
@BrandonEcholsMusic @haylau @Alive527 @akumar282 @kbuzsaki Thank you all so much for the help on this. My vision was to essentially replicate the way the BOTW bingo board worked but with TOTK content instead. What y'all have done has been super helpful. And I do believe the bingo board in up on bingosync. I saw SmallAnt and Linkus7 doing a lockout with it the other day. Thank you so much for the help and if you all want to add more goals or need to fix things with the current set of goals, feel free to reach out to me. You all have been amazing. Thank you so much!
-McBraggins

**Comment by akumar282** [comment]
> @BrandonEcholsMusic @haylau @Alive527 @akumar282 @kbuzsaki Thank you all so much for the help on this. My vision was to essentially replicate the way the BOTW bingo board worked but with TOTK content instead. What y'all have done has been super helpful. And I do believe the bingo board in up on bingosync. I saw SmallAnt and Linkus7 doing a lockout with it the other day. Thank you so much for the help and if you all want to add more goals or need to fix things with the current set of goals, feel free to reach out to me. You all have been amazing. Thank you so much!
> 
> -McBraggins

Np the pr needs to be merged for it to show up. I think what you saw was a custom board.

**Comment by McBraggins** [comment]
You are probably correct with the custom board thing. If we have code for all of this that we just made, I would like to have it so I can have a custom while the main website updates. I want to run this with my friend but I do not know a lot of the code that went into this. Thanks!

**Comment by BrandonEcholsMusic** [comment]
@McBraggins No problem! I'm so excited for it to be up and running!

**Comment by haylau** [comment]
I think we should leave the issue open until some list gets pushed? So we don't run into another scenario of more people trying to work on it? haha

**Comment by McBraggins** [comment]
I think closing lets people know that we figured it all out and we don’t need anymore help.

**Comment by haylau** [comment]
I disagree. Not until the game is on the website, otherwise it'll be difficult to find the work already done on TotK and might lead to another person opening a similar issue. The current PR is set to close this issue automatically when it gets merged.

**Comment by McBraggins** [comment]
That’s a valid point. I have reopened the issue.
